graveler/graveler.go

```diff
@@ -395,6 +395,9 @@ type CommittedManager interface {
 	// Get returns the provided key, if exists, from the provided TreeID
 	Get(ctx context.Context, ns StorageNamespace, treeID TreeID, key Key) (*Value, error)
 
+	// IsTreeExist returns true iff the tree with treeID exists
+	IsTreeExist(ctx context.Context, ns StorageNamespace, treeID TreeID) (bool, error)
```

Comment:
Think we should open GetTree and not IsTreeExist, expose an interface that can represent a Tree with ID() as a method will do better work.

Reply:
GetTree will read objects we don't want to read, this is the cheapest way of achieving the same.

Reply:
@itaiad200 get tree needs to read just the main part